Distribution/Server/Features/EditCabalFiles.hs

```diff
@@ -153,13 +152,10 @@ diffCabalRevisionsByteString oldRevision newRevision =
     -- make sure the parseSpecVer heuristic agrees with the full parser.
     -- Note that diffCabalRevisions parses the newRevision a second time.
     parseSpecVerCheck = case parseGenericPackageDescriptionChecked newRevision of
-       (_,ParseFailed err) -> Just $ showError (locatedErrorMsg err)
-       (True, ParseOk {})  -> Nothing
-       (False,ParseOk {})  -> Just "The 'cabal-version' field could not be properly parsed"
-
-    showError (Nothing, msg) = msg
-    showError (Just n, msg) = "line " ++ show n ++ ": " ++ msg
-
+       (True, _, Right _)      -> Nothing -- parsing successful
+       (_, _, Left (_, err:_)) -> Just $ showPError "" err -- TODO: show all errors
```

Comment:
`unlines` here too?

Reply:
I didn't because I'm not sure it would render sensible; note that this may be converted into HTML.

Reply:
:+1: